File: lib/yargui/frame.rb
# To change this template, choose Tools | Templates
# and open the template in the editor.
require 'java'


module Yargui
  class Frame < javax.swing.JFrame
    def initialize(w = 640, h= 480, name="Yargui")
      super(name)
      self.setSize(w, h)
    end

    # Adds a child component to this comonent's content pane
    def <<(child)
      self.getContentPane().add(child)
    end
  end

end
      

File: old/pxldrj/src/pxldrj/Frame.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package pxldrj;

import javax.swing.JComponent;
import javax.swing.JFrame;

/**
 *
 * @author bjmey
 */
public class Frame extends JFrame {
    Frame(String name, int w, int h) {
        super(name);
        this.setSize(w, h);
    }

    void addChild(JComponent child) {
        this.getContentPane().add(child);
    }

}
